lib/mosaic-utils/src/main/java/org/eclipse/mosaic/lib/util/gson/UnitFieldAdapter.java

```diff
@@ -79,7 +79,28 @@ private UnitFieldAdapter(boolean failOnError, Pattern pattern) {
 
     @Override
     public void write(JsonWriter out, Double param) throws IOException {
-        out.value(ObjectUtils.defaultIfNull(param, 0d));
+        String unit = null;
+        if(DISTANCE_PATTERN.equals(this.pattern)){
```

Comment:
I would suggest that the constructor defines which unit is used for export and your store in the UnitFieldAdapter. This way you do not rely on the large if-else-construct.

```java
    public static class DistanceMeters implements TypeAdapterFactory {
        @SuppressWarnings("unchecked")
        @Override
        public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
            return (TypeAdapter<T>) new UnitFieldAdapter(true, DISTANCE_PATTERN, "m");
        }
    }

    public static class SpeedMS implements TypeAdapterFactory {
        @SuppressWarnings("unchecked")
        @Override
        public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
            return (TypeAdapter<T>) new UnitFieldAdapter(true, SPEED_PATTERN, "m/s");
        }
```

Reply:
After fixing the tests I did. ;)